pkg/auth/handler/forgotpwd/reset_form.go

```diff
@@ -291,7 +291,21 @@ func (h ForgotPasswordResetFormHandler) ServeHTTP(rw http.ResponseWriter, r *htt
 	}
 
 	hashedPassword := principals[0].HashedPassword
-	email := templateCtx.user.LoginIDs["email"]
+
+	email := ""
+	for _, principal := range principals {
+		if principal.LoginIDKey == "email" {
```

Comment:
We cannot assume login id key`email` is email

Reply:
Yes, this will be changed in stage 2. I just keep the behavior same for now.